pkg/dockerregistry/server/prune/prune.go

```diff
@@ -19,6 +19,112 @@ import (
 	imageapiv1 "github.com/openshift/origin/pkg/image/apis/image/v1"
 )
 
+// Pruner defines a common set of operations for pruning
+type Pruner interface {
+	DeleteRepository(ctx context.Context, reponame string) error
+	DeleteManifestLink(ctx context.Context, svc distribution.ManifestService, reponame string, dgst digest.Digest) error
+	DeleteBlob(ctx context.Context, dgst digest.Digest) error
+}
+
+// DryRunPruner prints information about each object that going to remove.
+type DryRunPruner struct{}
+
+var _ Pruner = &DryRunPruner{}
+
+func (p *DryRunPruner) DeleteRepository(ctx context.Context, reponame string) error {
+	logger := context.GetLogger(ctx)
+	logger.Printf("Would delete repository: %s", reponame)
+	return nil
+}
+
+func (p *DryRunPruner) DeleteManifestLink(ctx context.Context, svc distribution.ManifestService, reponame string, dgst digest.Digest) error {
+	logger := context.GetLogger(ctx)
+	logger.Printf("Would delete manifest link: %s@%s", reponame, dgst)
+	return nil
+}
+
+func (p *DryRunPruner) DeleteBlob(ctx context.Context, dgst digest.Digest) error {
+	logger := context.GetLogger(ctx)
+	logger.Printf("Would delete blob: %s", dgst)
+	return nil
+}
+
+// RegistryPruner deletes objects.
+type RegistryPruner struct {
+	StorageDriver driver.StorageDriver
+}
+
+var _ Pruner = &RegistryPruner{}
+
+func (p *RegistryPruner) DeleteRepository(ctx context.Context, reponame string) error {
+	logger := context.GetLogger(ctx)
+	vacuum := storage.NewVacuum(ctx, p.StorageDriver)
+
+	logger.Debugln("Removing %s repository", reponame)
+	if err := vacuum.RemoveRepository(reponame); err != nil {
+		return fmt.Errorf("unable to remove the repository %s: %v", reponame, err)
+	}
+
+	return nil
+}
+
+func (p *RegistryPruner) DeleteManifestLink(ctx context.Context, svc distribution.ManifestService, reponame string, dgst digest.Digest) error {
+	logger := context.GetLogger(ctx)
+
+	logger.Printf("Deleting manifest link: %s@%s", reponame, dgst)
```

Comment:
not sure how you've made the decision between printf and debugf for the various Delete* functions, but it seems inconsistent.